src/pages/in/embed/nft/class/_classId.vue

```diff
@@ -0,0 +1,190 @@
+<template>
+  <NFTWidget
+    ref="widget"
+    :title="contentTitle"
+    :description="contentDescription"
+    :img-src="contentImage"
+    :url="contentURL"
+    :price="nftPrice"
+    :collector-count="nftCollectorCount"
+    :collected-count="nftCollectedCount"
+    :owner-address="iscnOwnerAddress"
+    :owner-display-name="iscnOwnerDisplayName"
+    :owner-avatar-src="iscnOwnerAvatarSrc"
+    :is-content-clickable="false"
+    @view-details="viewNFTDetails"
+    @collect="collectNFT"
+    @like="likeISCN"
+    @load-image="notifyParentOfResizing"
+  />
+</template>
+
+<script>
+import {
+  APP_LIKE_CO_URL_BASE,
+  LIKER_LAND_URL_BASE,
+} from '~/constant';
+import {
+  apiGetLikerDataByAddress,
+  apiGetNFTMetadata,
+  apiGetNFTOwners,
+  apiGetNFTPurchaseInfo,
+} from '~/util/api/api';
+import { getClassInfo } from '~/util/nft';
+
+export default {
+  layout: 'widget',
+  computed: {
+    widgetId() {
+      return this.$route.query.wid;
+    },
+    detailsURL() {
+      return `${LIKER_LAND_URL_BASE}/nft/class/${this.classId}`;
+    },
+    purchaseURL() {
+      return `${APP_LIKE_CO_URL_BASE}/nft/purchase/${encodeURIComponent(
+        this.iscnId
+      )}`;
+    },
+  },
+  async asyncData({ params }) {
+    const { classId } = params;
```

Comment:
I think in future likecoin-button use case, we only want user to input `iscn_id` and don't need to care about nft class id
So I imagine the flow would be `iscn_id` as param, fetch `/purchase` API to see if NFT is minted. If minted, the API would also return class id if you need it.
In short term, maybe we would want the nft widget to auto disable if `iscn_id` is not set or nft is not mint; and show this widget if NFT is minted (ie `classId` exists in api) via querying by `iscn_id`

Reply:
Or would this page be a used in a separated use case where we put int `class_id` only? Where would we use this?